examples/querying_entity_traits.ipynb

```diff
@@ -0,0 +1,710 @@
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "v1_ref = manager.createEntityReference(\"bal:///project_artwork/logos/openassetio?v=1\")\n",
+    "\n",
+    "trait_set_or_error = manager.entityTraits(\n",
+    "    logo_ref, EntityTraitAccess.kWrite, context, BatchElementErrorPolicyTag.kVariant)\n",
+    "\n",
+    "helpers.display_result(trait_set_or_error)"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "ee61dbb2bd4b828f"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Since the entity is read-only, we get an `EntityAccessError`."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "fff41a2190e269b6"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Generic resolve and re-publish\n",
+    "\n",
+    "A rare use-case is for a tool that blindly resolves, updates, and re-publishes an entity, regardless what type of entity it is. \n",
+    "\n",
+    "Such generic re-publishing is discouraged and dangerous. For example, if \"approval\" status is blindly resolved and re-published, the manager may not know how it should handle this properly. A better approach to such a generic tool is to present the traits and their properties to the user to select before re-publishing.\n",
+    "\n",
+    "However, such workflows are possible and have their place in a pipeline. This can be accomplished by making use of `entityTraits`. "
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "9c4274b9aa5f4965"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Re-publishing a mutated entity\n",
+    "\n",
+    "The following presents an example of blindly updating the display name for any entity. Note that we make use of the fact that managers should silently ignore data that it cannot persist when publishing."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "92a3d36ca0b4a816"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "# Get the complete trait set of the entity.\n",
+    "entity_trait_set = manager.entityTraits(logo_ref, EntityTraitAccess.kRead, context)\n",
+    "\n",
+    "# Ensure the manager will accept a publish of this entity with an updated display name.\n",
+    "policy_data = manager.managementPolicy(\n",
+    "    entity_trait_set | {DisplayNameTrait.kId}, PolicyAccess.kWrite, context)\n",
+    "\n",
+    "if not DisplayNameTrait.kId in policy_data.traitSet():\n",
+    "    raise Exception(\"Cannot update display name of this entity\")\n",
+    "\n",
+    "# Get all the properties of the given entity.\n",
+    "data_to_publish = manager.resolve(logo_ref, entity_trait_set, ResolveAccess.kRead, context)\n",
+    "\n",
+    "# Create/update the name of the entity.\n",
+    "DisplayNameTrait(data_to_publish).setName(\"My New Name\")\n",
+    "\n",
+    "# Publish it. Any properties we `resolve`d that cannot be re-published will be silently ignored.\n",
```

Comment:
This is missing re-imbuing any traits in `entity_trait_set` that didn't resolve as they dont have properties.

Reply:
Good spot, added in 81c1902f

If this becomes a common operation, we should add a convenience to imbue a whole trait set at once. But these examples are edge cases for now.

Reply:
:facepalm: 69957f8d